### PR TITLE
Add an NFData instance to HtmlT

### LINKED ISSUE #139: An NFData instance for HtmlT
Hi @chrisdone! I'm in the process of fixing some thunk retention in Flora and I wanted to know if you'd be interested in an NFData instance for HtmlT. :) 
I'd provide the PR of course.

### PR DESCRIPTION
**Pull request by tchoutri**
Fixes #139

### CODE REVIEW

File: lucid1/src/Lucid/Base.hs [hunk 1]
```diff
@@ -114,6 +115,9 @@ newtype HtmlT m a =
   deriving (Typeable)
 #endif
 
+instance NFData (HtmlT m a) where
```
[the remark this answers is not in the document]
Comment (tchoutri) [reply lucid1/src/Lucid/Base.hs:118]: Ah you're right I guess I was slightly confused. We could probably force the evaluation of everything?

[the remark this answers is not in the document]
Comment (chrisdone) [reply lucid1/src/Lucid/Base.hs:118]: If that’s what you need. It’s possible to have an instance that would render the thing to a bytestring and also deepseq the resulting `a`. But probably only where `m` is `Identity`. 🤔 Other mtl monad transformers don’t seem to have instances either, now that I look.

How about you add an orphan instance in your project, or use a patched fork, and continue your profiling efforts and if a certain instance ends up being valuable we can look at adding it?

[the remark this answers is not in the document]
Comment (tchoutri) [reply lucid1/src/Lucid/Base.hs:118]: Very well. :) I'll re-open the PR when I have something valuable

[the remark this answers is not in the document]
Comment (chrisdone) [reply lucid1/src/Lucid/Base.hs:118]: Cool! 👌

**Pull request closed without merging**